src/main/java/seedu/duke/Ui.java

```diff
@@ -0,0 +1,380 @@
+        System.out.println(greeting);
+    }
+
+    /**
+     * Prints the exception of loading data file into arraylist error.
+     */
+    public void showLoadingError() {
+        System.out.println("\u2639 OOPS!!! There is an error loading data file");
+    }
+
+    /**
+     * Returns the exception string for a command Duke does not understand.
+     *
+     * @return String of exception message when user input cannot be understood.
+     */
+    public String noSuchCommand() {
+        return "\u2639 OOPS!!! I'm sorry but I don't know what that means :-(\n";
+    }
+
+    /**
+     * Prints the message of the task that is deleted with its information.
+     *
+     * @param task Task that is deleted.
+     */
+    public void printDeletedTaskMsg(Task task) {
+        System.out.println("Noted. I've removed this task:\n" + task);
+    }
+
+    /**
+     * Prints the number of tasks in the list.
+     *
+     * @param tasks TaskList of the tasks in the list currently.
+     */
+    public void printNoOfTaskInList(TaskList tasks) {
+        String statusOfList;
+        if (tasks.size() == 1) {
+            statusOfList = "Now you have 1 task in the list.\n";
+        } else {
+            statusOfList = "Now you have " + tasks.size() + " tasks in the list.\n";
+        }
+        System.out.println(statusOfList);
+    }
+
+    /**
+     * Checks the exception for delete commands.
+     * Throws DukeException when the command is intended to delete a task but is
+     * incorrectly inputted by the user.
+     *
+     * @param command String of command that user input.
+     * @param tasks TaskList of all the tasks currently.
+     * @throws DukeException If there is a incorrectly inputted user command
+     * that is intended to delete task.
+     */
+    public void checkErrorForDeleteCommand(String command, TaskList tasks) throws DukeException {
+        if (command.contains(" ")) {
+            //throw exception for no task number and there is just trailing whitespaces
+            String res = command.replace(" ", "");
+            if (res.length() == 6) {
+                throw new DukeException("\u2639 OOPS!!! Please input the task number you would like to delete.\n");
+            }
+        } else if (command.length() == 6) {
+            //throw exception for no task number
+            throw new DukeException("\u2639 OOPS!!! Please input the task number you would like to delete.\n");
+        }
+        int curr = Parser.taskToDelete(command);
+        if (tasks.size() == 0) {
+            //check if list has no task to throw exception
+            throw new DukeException("\u2639 OOPS!!! You do not have any tasks in your list.\n");
+        } else if (curr > tasks.size()) {
+            //check if index is within list size or throw exception
+            throw new DukeException("\u2639 OOPS!!! You do not have that task in your list. "
+                    + "Call 'list' to see all your tasks :-)\n");
+        }
+    }
+
```

Comment:
Shouldn't the UI file handle only input and output? I think that perhaps error checking should occur during parsing, or be in the same class of the specific command itself.